### DISCUSSION TITLE
Ollama can't be conencted

### DISCUSSION THREAD

**Discussion by hellangleZ** [lobehub/lobehub#5664] [General Question] [answered]
### 📦 Environment

- [ ] Official
- [ ] Official Preview
- [ ] Vercel / Zeabur / Sealos
- [X] Docker
- [ ] Other

### 📌 Version

0.162.22/ latest

### 💻 Operating System

- [ ] Windows
- [ ] macOS
- [X] Ubuntu
- [ ] Other Linux
- [ ] iOS
- [ ] Android
- [ ] Other

### 🌐 Browser

- [X] Chrome
- [ ] Edge
- [ ] Safari
- [ ] Firefox
- [ ] Other

### 🐛 Bug Description

![image](https://github.com/lobehub/lobe-chat/assets/15274284/3ae1740b-120d-46ae-be61-eb4907165c8a)
Just configure by the guidance , but still couldn't conenct to ollama local

![image](https://github.com/lobehub/lobe-chat/assets/15274284/6eefcd61-0289-468a-8543-a7bfa2d08597)

![image](https://github.com/lobehub/lobe-chat/assets/15274284/9ca8d61c-6dea-451d-b160-942cbb5b9c53)

I try the curl , it could works at 127.0.0.1

![image](https://github.com/lobehub/lobe-chat/assets/15274284/83fb03ed-865c-4323-94f2-2e1449739bda)

ollama list

![image](https://github.com/lobehub/lobe-chat/assets/15274284/54095a6b-ac93-4570-97bf-ecdb76c06846)

[a comment by an automated account is not shown]

**Comment by mozillo** [comment]
https://github.com/lobehub/lobe-chat/issues/2855 CORS error solution

**Comment by hellangleZ** [comment]
> #2855 CORS error solution
 @mozillo 
Would you please check my configuration, such as 
![image](https://github.com/lobehub/lobe-chat/assets/15274284/d5afa64f-5cfe-4f4f-b0d3-cc3eda9f834f)

**Comment by 745854919** [comment]
我也是有同样的问题，用反代域名在浏览器里可以连通，但是始终是不如内网直接连接更安全方便些

[a comment by an automated account is not shown]

**Comment by 9Somboon** [comment]
same problem.

**Comment by ShravanSunder** [comment]
Still having the same problem after setting cors

**Comment by ktneely** [comment]
I'm seeing this, as well, and it appears to be that the platform is not honoring the Ollama settings.  On a new install, I have $PROXY_URL and $OLLAMA_MODEL_LIST specified in the `.env` file and referenced by `docker-compose.yml`, just as for other inference servers.  OpenAI, as an example, works fine, but the Ollama config shows defaults instead of the environment settings.  

If I manually configure Ollama in the browser, it works.

**Comment by oatmealm** [comment]
same problem nothing helps, localhost, 127.0.0.1, host.docker.internal:11434, network_mode host... it won't connect

**Comment by tuxthepenguin84** [comment]
I'm having this issue as well and haven't been able to fix it, however it appears Lobe-Chat is using method OPTIONS, when it should be using GET (?)

```
[GIN] 2024/12/06 - 18:42:15 | 403 |       18.23µs |   192.168.99.11 | OPTIONS  "/api/tags"
[GIN] 2024/12/06 - 18:42:19 | 200 |    1.255754ms |   192.168.99.11 | GET      "/api/tags"
[GIN] 2024/12/06 - 18:42:30 | 403 |       16.79µs |   192.168.99.11 | OPTIONS  "/api/tags"
```

I have no issues connecting to my ollama server with Open-WebUI and it uses GET not OPTIONS

**Comment by tuxthepenguin84** [comment, marked as answer]
OK I figured it out on my end. I run ollama and lobe-chat behind an nginx proxy.

I needed to make two configuration changes:

In my ollama docker-compose configuration I needed to add the following:
```
environment:
      OLLAMA_ORIGINS: '*'
```

Then in my nginx docker-compose configuration, I had
`add_header Referrer-Policy         "no-referrer-when-downgrade" always;`
however I had to comment that out.

Once those two configs were in place everything worked in Lobe-Chat for Ollama.

**Comment by sushibait** [comment]
> OK I figured it out on my end. I run ollama and lobe-chat behind an nginx proxy.
> 
> I needed to make two configuration changes:
> 
> In my ollama docker-compose configuration I needed to add the following:
> 
> ```
> environment:
>       OLLAMA_ORIGINS: '*'
> ```
> 
> Then in my nginx docker-compose configuration, I had `add_header Referrer-Policy "no-referrer-when-downgrade" always;` however I had to comment that out.
> 
> Once those two configs were in place everything worked in Lobe-Chat for Ollama.

Would you mind sharing your nginx conf? I am experiencing the exact same issue.

**Comment by tuxthepenguin84** [comment]
Here's part of it:

```
add_header X-XSS-Protection        "1; mode=block" always;
add_header X-Content-Type-Options  "nosniff" always;
#add_header Referrer-Policy         "no-referrer-when-downgrade" always;
add_header Permissions-Policy      "interest-cohort=()" always;

location ~ /\.(?!well-known) {
  deny all;
}
```
